generate_test_coverage.sh

```diff
@@ -40,71 +82,71 @@ if ! grcov --version &>/dev/null; then
-    exit_with_error "Coverage run of cargo test failed." "$TEST_JSON_OUTPUT"
-fi
-echo "Finished cargo test successfully"
-cargo profdata -- merge -sparse uhyve-*.profraw -o uhyve.profdata \
-    || exit_with_error "Failed to merge raw profiling data"
-
 # Generate code coverage in coveralls JSON format
 grcov "$DIR" --source-dir "$DIR" \
     --binary-path "$DIR/target/debug" \
     --output-type coveralls \
     --token "$UHYVE_CODECOV_TOKEN" > coveralls.json \
  || exit_with_error "grcov did not successfully generate a coverage report"
 
-# Todo: make this whole part optional, since we might be only interested in coveralls report
-# Remove non JSON parts. Assumes all json lines start with '{' (valid for cargo output as of now)
-# but this may break sometime.
-FILTERED_TEST_JSON_OUTPUT=$(grep '^[\{]' <<<"$TEST_JSON_OUTPUT")
-# Get test executable names (adapted from https://doc.rust-lang.org/nightly/unstable-book/compiler-flags/source-based-code-coverage.html)
-JQ_OUTPUT=$(jq -r "select(.profile.test == true) | .filenames[]" <<<"$FILTERED_TEST_JSON_OUTPUT")
-JQ_RETURN=$?
-TEST_FILES=$(grep -v dSYM - <<<"$JQ_OUTPUT")
-GREP_RETURN=$?
-if [ $JQ_RETURN != 0 ]; then
-    jq_input=$(mktemp)
-    jq_output=$(mktemp)
-    echo "$FILTERED_TEST_JSON_OUTPUT" >"$jq_input"
-    echo "$JQ_OUTPUT" >"$jq_output"
-    echo "Error: using jq to parse test executable names from cargo output."
-    echo "The input json was dumped to  'file://$jq_input'."
-    echo "The Output of jq was dumped  'file://$jq_output'."
-    exit 1
-fi
-if [ $GREP_RETURN != 0 ]; then
-    echo "Grep reported an error - dumping output:"
-    echo "$TEST_FILES"
-    exit 1
+echo "generated coveralls.json"
+
+
+if [ "$PRINT_COVERAGE" = true ]; then
+    # Remove non JSON parts. Assumes all json lines start with '{' (valid for cargo output as of now)
+    # but this may break sometime.
+    FILTERED_TEST_JSON_OUTPUT=$(grep '^[\{]' <<<"$TEST_JSON_OUTPUT")
+    # Get test executable names (adapted from https://doc.rust-lang.org/nightly/unstable-book/compiler-flags/source-based-code-coverage.html)
+    JQ_OUTPUT=$(jq -r "select(.profile.test == true) | .filenames[]" <<<"$FILTERED_TEST_JSON_OUTPUT")
+    JQ_RETURN=$?
+    TEST_FILES=$(grep -v dSYM - <<<"$JQ_OUTPUT")
+    GREP_RETURN=$?
+    if [ $JQ_RETURN != 0 ]; then
+        jq_input=$(mktemp)
+        jq_output=$(mktemp)
+        echo "$FILTERED_TEST_JSON_OUTPUT" >"$jq_input"
+        echo "$JQ_OUTPUT" >"$jq_output"
+        echo "Error: using jq to parse test executable names from cargo output."
+        echo "The input json was dumped to  'file://$jq_input'."
+        echo "The Output of jq was dumped  'file://$jq_output'."
+        exit 1
+    fi
+    if [ $GREP_RETURN != 0 ]; then
+        echo "Grep reported an error - dumping output:"
+        echo "$TEST_FILES"
+        exit 1
+    fi
+    echo "Info: Detected the following Test executables: $TEST_FILES"
+
+    DOC_TEST_BINS=("$DIR/target/debug/doctestbins/*/rust_out")
```

Comment:
Here a check needs to be added if the doc was built at all. Otherwise the expression won't expand and cargo-cov will fail later on since `/target/debug/doctestbins/*/rust_out` is not a file that exists.

Reply:
Resolved by magic shell fuckery